internal/prober/multihttp/script.go

```diff
@@ -17,13 +18,37 @@ import (
 //go:embed script.tmpl
 var templateFS embed.FS
 
-func buildUrl(req *sm.MultiHttpEntryRequest) string {
-	// If we are here, the request has already been validated, and the URL
-	// should be valid. This function should never return an error.
-	u, _ := url.Parse(req.Url)
+func replaceVariablesInString(s string) string {
+	re := regexp.MustCompile(`\$\{(.*?)\}`)
+	variableNames := re.FindAll([]byte(s), -1)
+	transformed := s
+	if len(variableNames) > 0 {
+		for _, variableName := range variableNames {
+			variableString := string(variableName)
+			strippedVarName := strings.Replace(variableString, "${", "", 1)
+			strippedVarName = strings.Replace(strippedVarName, "}", "", 1)
+			var replacement strings.Builder
+			replacement.WriteString("${vars['")
+			replacement.WriteString(strippedVarName)
+			replacement.WriteString("']}")
+			transformed = strings.Replace(transformed, string(variableString), replacement.String(), -1)
+		}
+	}
+	return transformed
+}
 
+func buildUrl(req *sm.MultiHttpEntryRequest) string {
 	var buf strings.Builder
+	url := replaceVariablesInString(req.Url)
+	buf.WriteString("new URL(`")
+	buf.WriteString(url)
+	buf.WriteString("`)")
+	return buf.String()
+}
 
+func buildUrlString(req *sm.MultiHttpEntryRequest) string {
```

Comment:
This exists only to provide something for `check` to print in a way that the plugin can query reliably